libs/common/src/middleware/feature-flags/validate-feature-flag-enabled.ts

```diff
@@ -0,0 +1,45 @@
+import { RequestHandler } from 'express';
+import { HttpStatusCode } from 'axios';
+import { isPortalFacilityAmendmentsFeatureFlagEnabled, isTfmDealCancellationFeatureFlagEnabled } from '../../helpers';
+
+/**
+ * Creates a middleware function to check if the given feature flag is enabled
+ *
+ * This will send status `404` if the feature flag is disabled and call `next` otherwise
+ */
+const generateBackendFeatureFlagMiddleware =
+  (isFeatureFlagEnabled: () => boolean): RequestHandler =>
+  (req, res, next) => {
+    if (!isFeatureFlagEnabled()) {
+      console.info(`Feature flag disabled, accessing ${req.originalUrl}`);
```

Comment:
Is there a way we can consume and log out the feature flag name here? Feels cryptic at the moment. 

```ts
console.info(`Feature flag ${featureFlagName} disabled whilst accessing ${req.originalUrl}`);
```